blob/singularity_store.go

```diff
@@ -22,13 +27,15 @@ const maxCarSize = "31.5GiB"
 type SingularityStore struct {
 	local             *LocalStore
 	sourceID          uint32
+	wallet            *wallet.Wallet
 	singularityClient client.Client
 }
 
-func NewSingularityStore(dir string, singularityClient client.Client) *SingularityStore {
+func NewSingularityStore(dir string, wallet *wallet.Wallet, singularityClient client.Client) *SingularityStore {
```

Comment:
Alternatives I can think of are:
* Move all the config into a package and pass that around
  * pro: simpler
  * con: not all stores have the same config, meaning we end up with irrelevant config for some stores.
* Take a storeCreator as an option in Motion and let store define its own options.
  * pro: each store has full control over its options
  * con: more boilerplate.